packages/react-components/src/components/timegraph-output-component.tsx

```diff
@@ -521,6 +543,101 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
         );
     }
 
+    onCtxMenu(event: React.MouseEvent<HTMLDivElement>): void {
+        event.preventDefault();
+        event.stopPropagation();
+        const refNode = this.treeRef.current;
+        let calculatedX = 0;
+        let calculatedY = 0;
+        if (refNode) {
+            calculatedX = event.clientX;
+            calculatedY = event.clientY - refNode.getBoundingClientRect().top;
+        }
+
+        const { show } = useContextMenu({
+            id: MENU_ID + this.props.outputDescriptor.id
+        });
+
+        show(event, {
+            props: {},
+            position: { x: calculatedX, y: calculatedY }
+        });
+    }
+
+    protected handleItemClick = (params: ItemParams): void => {
+        const props = {
+            outputDescriptor: this.props.outputDescriptor,
+            selectedRow: this.state.selectedRow,
+            selectedRows: this.state.selectedRows
+        };
+        const signalPayload: ContextMenuItemClickedSignalPayload = new ContextMenuItemClickedSignalPayload(
+            params.data.itemId,
+            params.data.menuId,
+            props
+        );
+        signalManager().fireContextMenuItemClicked(signalPayload);
+    };
+
+    renderContextMenu(): React.ReactNode {
+        return (
```

Comment:
right mouse click on a row of a timegraph view that doesn't have a context menu registered I get the following error in the console log (descriptor id is the one I used and will vary for other data providers):
```
root ERROR It seems that the menu you are trying to display is not renderer or you have a menu id mismatch. You used the menu id: timegraph.context.menuId org.eclipse.tracecompass.incubator.internal.callstack.core.instrumented.provider.flamechart:org.eclipse.tracecompass.incubator.callstack.core.lttng.ust```

Reply:
Ah - I forgot to add a check in onCtxMenu to see if a context menu was contributed - will fix in next patch!